[iOS] Unable to show the software keyboard when programmatically focusing a text field during `dblclick`

aprotyas opened the pull request:
#### 45fc397579fade6934aeab0bba233006f550580d
<pre>
[iOS] Unable to show the software keyboard when programmatically focusing a text field during `dblclick`
<a href="https://bugs.webkit.org/show_bug.cgi?id=253991">https://bugs.webkit.org/show_bug.cgi?id=253991</a>
rdar://104600783

Reviewed by Wenson Hsieh.

This commit adds the `dblclick` event to the set of events for which we
allow the software keyboard to be shown programmatically, which means
that programmatic focus now shows the software keyboard underneath a
user&apos;s double click interaction.

* LayoutTests/fast/forms/ios/show-software-keyboard-on-programmatic-focus-during-dblclick-expected.txt: Added.
* LayoutTests/fast/forms/ios/show-software-keyboard-on-programmatic-focus-during-dblclick.html: Added.
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::handleDoubleTapForDoubleClickAtPoint):

Canonical link: <a href="https://commits.webkit.org/261751@main">https://commits.webkit.org/261751@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/14d54a8425571114d71c3d2cb970d4518877b94a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112667 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21821 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/1337 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4441 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/121194 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/23162 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/12986 "Built successfully") | [⏳ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/WPE-WK2-Tests-EWS "Waiting to run tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118435 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/17205 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100433 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105743 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/99157 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/978 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/46218 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/14144 "Built successfully") | [⏳ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/95420 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14830 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/10372 "2 flakes 4 failures") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/20161 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/53012 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16674 "Built successfully") | | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/4491 "Failed to push commit to Webkit repository") | | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on LayoutTests/fast/forms/ios/show-software-keyboard-on-programmatic-dblclick-of-text-field.html:
```diff
@@ -0,0 +1,36 @@
+jsTestIsAsync = true;
+
+addEventListener("load", async () => {
+    description("Tests that a software keyboard is shown when a text field is progammatically double clicked.");
```
I think this test currently works by showing the hardware keyboard, rather than the software keyboard.

We could either adjust the description, or add this before calling `UIHelper.doubleActivateAt()`:
```
    await UIHelper.setHardwareKeyboardAttached(false);
```

whsieh replied:
I would also name the test something along the lines of:
> show-software-keyboard-on-programmatic-focus-during-dblclick.html

instead of:

> show-software-keyboard-on-programmatic-dblclick-of-text-field.html

aprotyas replied:
Detaching the hardware keyboard seems more intuitive to me -- fixed in https://github.com/WebKit/WebKit/commit/14d54a8425571114d71c3d2cb970d4518877b94a

aprotyas replied:
Renamed in the same commit https://github.com/WebKit/WebKit/commit/14d54a8425571114d71c3d2cb970d4518877b94a

whsieh reviewed (approved):
r=me; let's wait for iOS EWS to finish before landing.

aprotyas committed:
[iOS] Unable to show the software keyboard when programmatically focusing a text field during `dblclick`
https://bugs.webkit.org/show_bug.cgi?id=253991
rdar://104600783

Reviewed by Wenson Hsieh.

This commit adds the `dblclick` event to the set of events for which we
allow the software keyboard to be shown programmatically, which means
that programmatic focus now shows the software keyboard underneath a
user's double click interaction.

* LayoutTests/fast/forms/ios/show-software-keyboard-on-programmatic-focus-during-dblclick-expected.txt: Added.
* LayoutTests/fast/forms/ios/show-software-keyboard-on-programmatic-focus-during-dblclick.html: Added.
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::handleDoubleTapForDoubleClickAtPoint):

Canonical link: https://commits.webkit.org/261751@main
